### DISCUSSION TITLE
All users sharing the same history/memory

### DISCUSSION THREAD

**Discussion by magnum6actual** [FlowiseAI/Flowise#829] [Q&A]
I'm just getting going with Flowise. I've deployed to GCP and have been using a model endpoint from a web app. I've noticed that the memory of the model Is shared among users. Which I guess would make sense as I don't see anything that would indicate sessions being created.  So - how would one handle that?  I clearly don't want different users to be able to query other's histories. And, if more than just a few people are using an app pointing to the endpoint, memory becomes chaos.

I'm assuming I've missed something simple on the setup or deployment as I'm sure no one intends to have shared memory among all users on a model.

**Comment by magnum6actual** [comment]
I think I answered my own question. I need to be using Zep memory for an API with multiple users.

https://docs.flowiseai.com/memory/zep-memory

**Reply by chungyau97** [reply]
Hi @magnum6actual,

You also need to set unique sessionId for chat embed if not it will be using [chatflowid Line: 810-811](https://github.com/FlowiseAI/Flowise/blob/main/packages/server/src/index.ts#L810)

**Reply by LeandroTecware** [reply]
Is there any way to solve this without using Zep Memory? If I generate a SessionId and send it through the API as overrideConfig will the system produce different sessions?
